src/main/scala/nl/knaw/dans/easy/license/package.scala

```diff
@@ -15,19 +15,241 @@
+        case x => x
+      }
+    }
+
+    def doOnSuccess(f: T => Unit): Try[T] = {
+      t match {
+        case Success(x) => Try { f(x); x }
+        case e => e
+      }
+    }
+
+    def eventually[Ignore](effect: () => Ignore): Try[T] = {
+      t.doOnSuccess(_ => effect()).doOnError(_ => effect())
+    }
+
+    def toObservable: Observable[T] = {
+      t match {
+        case Success(x) => Observable.just(x)
+        case Failure(e) => Observable.error(e)
+      }
+    }
+  }
+
+  implicit class StringExtensions(val s: String) extends AnyVal {
+    /**
+      * Checks whether the `String` is blank
+      * (according to [[org.apache.commons.lang.StringUtils.isBlank]])
+      *
+      * @return
+      */
+    def isBlank = StringUtils.isBlank(s)
 
+    /** Converts a `String` to an `Option[String]`. If the `String` is blank
+      * (according to [[org.apache.commons.lang.StringUtils.isBlank]])
+      * the empty `Option` is returned, otherwise the `String` is returned
+      * wrapped in an `Option`.
+      *
+      * @return an `Option` of the input string that indicates whether it is blank
+      */
+    def toOption = if (s.isBlank) Option.empty else Option(s)
+  }
+
+  implicit class NamingEnumerationToObservable[T](val enum: NamingEnumeration[T]) extends AnyVal {
+    def toObservable = Observable.from(new Iterable[T] {
+      def iterator = new Iterator[T] {
+        def hasNext = enum.hasMore
+
+        def next() = enum.next()
+      }
+    })
+  }
+
+  implicit class CloseableResourceExtensions[T <: Closeable](val resource: T) extends AnyVal {
+    def usedIn[S](observableFactory: T => Observable[S], dispose: T => Unit = _ => {}, disposeEagerly: Boolean = false): Observable[S] = {
+      Observable.using(resource)(observableFactory, t => { dispose(t); t.closeQuietly() }, disposeEagerly)
+    }
+
+    def use[S](f: T => S, dispose: T => Unit = _ => {}): Try[S] = {
+      Try(f(resource)).eventually(() => { dispose(resource); resource.closeQuietly() })
+    }
+
+    def closeQuietly() = IOUtils closeQuietly resource
+  }
+
+  implicit class InputStreamExtensions(val stream: InputStream) extends AnyVal {
+    def loadXML = XML load stream
+  }
+
+  // TODO not used here anymore, but useful for debugging purposed. Maybe we can migrate this to a EASY-Utils project?
+  implicit class ObservableDebug[T](val observable: Observable[T]) extends AnyVal {
+    def debugThreadName(s: String = "") = observable.materialize.doOnEach(_ => println(s"$s: ${Thread.currentThread().getName}")).dematerialize
+    def debug(s: String = "") = observable.materialize.doOnEach(x => println(s"$s: $x")).dematerialize
+  }
+
+  def queryFedora[T](datasetID: DatasetID, datastreamID: String)(f: InputStream => T)(implicit client: FedoraClient): Observable[T] = {
+    FedoraClient.getDatastreamDissemination(datasetID, datastreamID)
+      .execute(client)
+      .getEntityInputStream
+      .usedIn(f.andThen(Observable.just(_)))
+  }
```

Comment:
To call execute without a client instance (which you have to drag around) StageDataset does the following:

    def setFedoraConnectionSettings(fedoraServiceUrl: String, fedoraUser: String, fedoraPassword: String): Unit = {
      val credentials = new FedoraCredentials(fedoraServiceUrl, fedoraUser, fedoraPassword)
      FedoraRequest.setDefaultClient(new FedoraClient(credentials))
    }